src/lib/multicall/Multicall.ts

```diff
@@ -116,13 +120,39 @@ export class Multicall {
     });
   }
 
-  viemClient: ReturnType<typeof Multicall.getViemClient>;
+  fallbackRpcSwitcher = new SlidingWindowFallbackSwitcher({
+    fallbackTimeout: 60 * 1000, // 1 minute
+    restoreTimeout: 5 * 60 * 1000, // 5 minutes
+    eventsThreshold: 3,
+    onFallback: () => {
+      getStaticOracleKeeperFetcher(this.chainId).fetchPostEventReport({
+        event: `multicall.${isWebWorker ? "worker" : "mainThread"}.fallbackRpcMode.on`,
+      });
+    },
+    onRestore: () => {
+      getStaticOracleKeeperFetcher(this.chainId).fetchPostEventReport({
+        event: `multicall.${isWebWorker ? "worker" : "mainThread"}.fallbackRpcMode.off`,
+      });
+    },
+  });
+
+  getClient: (options?: { forceFallback?: boolean }) => ReturnType<typeof Multicall.getViemClient>;
 
   constructor(
     public chainId: number,
-    public rpcUrl: string
+    public rpcUrl: string,
+    public fallbackRpcUrl?: string
```

Comment:
I don't see why fallbackRpcUrl can be undefined, you always provide it.
let's then not support undefined here and fallbackClient variable?